features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -230,9 +277,14 @@ internal class Base64Serializer private constructor(
 
     private fun finalizeRecordedDataItem(
         base64String: String,
+        hash: String,
         wireframe: MobileSegment.Wireframe.ImageWireframe,
         base64SerializerCallback: Base64SerializerCallback?
     ) {
+        if (hash.isNotEmpty()) {
+            wireframe.resourceId = hash
```

Comment:
So what backend does if there is no `resourceId`, but other payload like `base64`, etc. is there. How `resourceId` is used?

Reply:
at the moment backend is using base64, and eventually will map rssId as the key to pull the binary for use in the replays. In the future when the rss endpoint is done, there will have to be some logic there for ending support for b64 and moving over entirely to rssId (we'll start to send b64 as an empty field in the wireframes and only populate rssId)

Reply:
So what will be our logic then if we cannot generate `resourceId`? We won't send a resource itself then? This 99.999% won't happen (all devices for sure can generate MD5 hash), but I'm just curious.

Reply:
If we can't generate the hash then I think we'll need to throw away the binary because backend won't be able to use it in a replay. We can check for this in the future by having the rssId inside the recordedDataQueueItem holding the binary and marking it as invalid if the field isn't populated, then the dataQueueHandler can purge the record